authnid/token.py

```diff
@@ -0,0 +1,32 @@
+import time
+import hashlib
+import hmac
+import base64
+import binascii
+
+class TokenManager():
+    def __init__(self, secret):
+        self._secret = secret.encode()
+
+    # stub user ID until we implement it
+    def token(self, user_id=10000):
+        payload = f'{user_id}:{self._timestamp()}'
+        return self._encode(payload)
+
+    def validate(self, token):
+        try:
+            decoded = base64.urlsafe_b64decode(token).decode()
+            parts = decoded.split(':', 1)
+            recoded = self._encode(parts[1])
+            return hmac.compare_digest(token, recoded)
```

Comment:
In addition to checking the hmac, I think you also want to append the time to the token, and check it against the current time.

Reply:
Concur. These tokens should only be good for a few minutes at most... just long enough for someone on a slow connection to get redirect to ATST, and then have ATST issue a validation call.